main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
In general, I think we are in agreement that we should try to use the Scala identifier for the references (e.g. `root`) where possible. It's just that it would require additional knowledge of where the "current" project is to be able to display the relative names.

> I really don't see why we have to do this.

Why am I doing the `ProjectRef(uri("file:/xxx/helloworld/"),"root")` is simple. The general form of the sbt shell notation is [`{<build-uri>}<project-id>/config:intask::key`][1], and I wanted to make sure that the full gamut is expressible with slash syntax.

  [1]: http://www.scala-sbt.org/0.13/docs/Scopes.html#Referring+to+scoped+keys+from+the+sbt+shell